### PR TITLE
[WIP] Add charts to active admin dashboard

### PR DESCRIPTION
**Pull request by egemen-dev**
> It would be awesome to visualize some data from this application in active admin dashboard, such as adoptions, user sign ups and donations over time.

_Related issue:_ https://github.com/kasugaijin/baja-pet-rescue/issues/168

---

This PR is pretty self-descriptive as the title suggests. So, I think it does not need much explanation.

In this PR, I added the `chartkick` gem to generate charts in `activeadmin`.

There are 3 new sections in /admin/dashboard. `Adoptions Over Time`, `Donations Over Time` and `User Sign Ups Over Time`. They are all line charts. 

**_Example screenshot_**

Screenshot 2023-07-21 at 20-09-17 Dashboard Baja Pet Rescue

### CODE REVIEW

File: app/admin/dashboard.rb [hunk 1]
```diff
@@ -33,6 +33,21 @@
             end
           end
         end
+        # Chartkick charts here
+        panel "Adoptions Over Time" do
+          line_chart Adoption.all.group_by { |adp| adp.created_at.beginning_of_month }
```
Comment (egemen-dev) [inline app/admin/dashboard.rb:38]: Suggestion: We can use this https://github.com/ankane/groupdate gem to simplify the logic.

But, I remember that you don't want to add a gem if we can do without it. That is why it's the pure Rails right now.

**Review by egemen-dev — COMMENTED** [review]
You can merge it if you want. But, I put [WIP] on title to indicate that we can talk about it further. Also, it's open to any suggestions.

Note: I have no idea why Ruby on Rails CI / test (pull_request) fails on 'Install Ruby and gems' section.

**Comment by kasugaijin** [comment]
> Note: I have no idea why Ruby on Rails CI / test (pull_request) fails on 'Install Ruby and gems' section.

Could you please try committing `gemfile.lock` as it's the new chartkick gem that is failing on bunde install in the test container.

File: config/initializers/active_admin.rb [hunk 2]
```diff
@@ -3,9 +3,11 @@
   #
   # Set the title that is displayed on the main layout
   # for each of the active admin pages.
-  #
   config.site_title = "Baja Pet Rescue"
 
+  # Load the Google API JS library
+  config.register_javascript 'https://www.google.com/jsapi'
```
Comment (kasugaijin) [inline config/initializers/active_admin.rb:9]: Could you tell me a bit more how this gets used?

File: app/admin/dashboard.rb (hunk 1 above)
Comment (kasugaijin) [reply app/admin/dashboard.rb:38]: this looks good as is 👍

**Comment by egemen-dev** [comment]
> > Note: I have no idea why Ruby on Rails CI / test (pull_request) fails on 'Install Ruby and gems' section.
> 
> Could you please try committing `gemfile.lock` as it's the new chartkick gem that is failing on bunde install in the test container.
> 
> If that does not work, the solution probably lies here: https://stackoverflow.com/questions/72331753/ruby-and-rails-github-action-exit-code-16

I remember that you once told don't commit `Gemfile.lock`, am I wrong? That's why I did not :D Apparently, it solved the issue.

**Comment by kasugaijin** [comment]
> > > Note: I have no idea why Ruby on Rails CI / test (pull_request) fails on 'Install Ruby and gems' section.
> > 
> > 
> > Could you please try committing `gemfile.lock` as it's the new chartkick gem that is failing on bunde install in the test container.
> > If that does not work, the solution probably lies here: https://stackoverflow.com/questions/72331753/ruby-and-rails-github-action-exit-code-16
> 
> I remember that you once told don't commit `Gemfile.lock`, am I wrong? That's why I did not :D Apparently, it solved the issue.

Haha yes I was wrong in saying that. It was something I had learned not to do at my last job, but after a lot of googling it looks like gemfile.lock should always be committed. I have changed the readme to reflect that also.

File: config/initializers/active_admin.rb (hunk 2 above)
Comment (egemen-dev) [reply config/initializers/active_admin.rb:9]: This is for registering Google Chart JavaScript to active_admin.  Chartkick gem is a wrapper for several of the most popular open source chart libraries like Google Chart and Chart.js. Without this line, it does not work.

Comment (kasugaijin) [reply config/initializers/active_admin.rb:9]: Oh I see, ok that makes sense. Thank you.

**Pull request merged**